### PR TITLE
Avoid using MemBatch in `ValidateChain`

### PR DESCRIPTION
**Pull request by Giulio2002**
We take advantage of sharedDoms

### CODE REVIEW

File: turbo/execution/eth1/forkchoice.go [hunk 1]
```diff
@@ -284,10 +283,6 @@ func (e *EthereumExecutionModule) updateForkChoice(ctx context.Context, original
 				return
 			}
 		}
-		if e.executionPipeline.HasUnwindPoint() {
-			unwindToGenesis = e.executionPipeline.UnwindPoint() == 0
-		}
```
Comment (Giulio2002) [inline turbo/execution/eth1/forkchoice.go:289]: this is useless btw

File: erigon-lib/state/state_changeset.go [hunk 2]
```diff
@@ -251,3 +248,68 @@ func DeserializeKeys(in []byte) [kv.DomainLen][]DomainEntryDiff {
 	}
 	return ret
 }
+
+const diffChunkLen = 8*1024 - 32
```
Comment (Giulio2002) [inline erigon-lib/state/state_changeset.go:252]: had to move it here so that I can incorporate it with Flushing of domains

**Review by AskAlexSharov — CHANGES_REQUESTED** [review]
I don't understand how ValidateChain can do `BeginRw`. 
What if node does execution at this time? 
It means we have less time for pruning? 
What if you need validate 2 forks? 
Or did you mean `BeginRo`?
Plz add this info in description of PR, otherwise hard to understand the intent.

**Comment by Giulio2002** [comment]
Ok, so the "Write" part of RwTx is only used for unwinding beforehand. If you look at the code the RwTx is rollbacked right after `ValidatePayload`. It does not mean we have less time for pruning the flushing is done through `SharedDomains` we keep it and use that to `Flush`, so functionally, it is as if we had diff like before but in the form of SharedDomain's writers. if you need to validate 2 big forks, you can do that with no issue, because it is functionally equivalent to the previous one.

Regarding intent, Intent are:
*  to remove the `FlushExtendingFork` timing, before we were flushing domains on membatch and then flush membatch onto a RwTx, this was quite expensive in terms of performance
* Make the routine more stable, MemoryMutation is quite buggy, so it removes one of the things we have to worry about.

**Comment by awskii** [comment]
So this issue fixes DupSort issue in MemBatch causing
`[5/7 Execution] ExecV3: loadIntoTable : fn error: prune history commitment got invalid txNum: found 2434374999 != 2433726825 wanted"`

**Comment by Giulio2002** [comment]
yes that too

**Review by AskAlexSharov — APPROVED** [review]

**Pull request merged**